### ISSUE TITLE
Update Web-Api to dotnet core 3.1

### ISSUE THREAD

**Issue by SimonAnnetts** [jncc/web-mapper-core#119]
There might be some issues around finding the right nuget package versions and JSON libraries.

I will be sticking with Newtonsoft.Json as it supports serialising models that have a circular or semi circular structure. This is something that in core 3.1 the new MS System.Text.Json does not support yet.

**Comment by SimonAnnetts** [comment]
only basic testing of the API and crud pages done so far.

**Comment by SimonAnnetts** [comment]
Our test server will need updating to the dotnet core 3.1 SDK before I can push the build to that.

**Comment by SimonAnnetts** [comment]
Actually, I will be building a new dev server for this in our London data-centre (the existing dev server is in France). The implications of this will be that we will no longer be able to push code updates from our dev server to yours, but I think we agreed at the last meeting that this was no longer necessary because you'd be building from source with a CI process connected to Github ??

**Comment by mattdebont** [comment]
Do we have an idea on how we are working with branches on this r.e. MPA mapper / Emodnet mapper splits

**Comment by HelenwoodsJNCC** [comment]
@SimonAnnetts just to confirm we are happy for this to go ahead under this years support contract. If you can just let us know once  the updates are done and they are ready to be deployed.

**Comment by andyb-esdm** [comment]
@mattdebont there are two branches of interest
master - which is where all core functionality is implemented and represents the EMODNet deployment
analytics - this includes all master functionality (I merge it in) plus the google analytics scripts on index.html for the MPA mapper

**Comment by andyb-esdm** [comment]
@HelenwoodsJNCC I was asked to update the Angular framework too so I'll jump on to this issue too if that's okay.
I've upgraded Angular from v9 to v11 (via v10 as recommended) and tested in ie11 (still works!)
The latest version is v12 but ie11 support starts to become deprecated here and ie11 will not be supported in v13. I don't want to risk that right now given the problems we had with ie11 during initial release. v11 is under long term support until May 2022.
On that note it would be good to find out if Defra staff (and other agencies) are still using ie11. I know a lot of NE staff are now using Chrome for other projects.
Work so far has been 2.5h. I'd like to upgrade the slider and autocomplete libraries too. There will be a small amount of manual editing to do that but should take no longer than an hour more. Let me know if that's okay.

**Comment by GMDuncan** [comment]
@andyb-esdm Happy for you to spend the extra hour.

**Comment by GMDuncan** [comment]
Hi @andyb-esdm just checking how these updates are getting on.

**Comment by andyb-esdm** [comment]
@GMDuncan API has been updated to .net core 3.1 and angular updated to v11
I haven't spent that hour updating the two components that I should (the slider and select). I think @SimonAnnetts needs to upgrade our staging server to 3.1 too.
Are you waiting on us for this? If so, I'll coordinate with Simon and find time to finish off in the next few days.

**Comment by GMDuncan** [comment]
@andyb-esdm we're hoping to do a deployment to live by end of June if possible (including the new geoserver), and it'd be good to roll everything out together.

If it's possible to get this done shortly, it'd be grand :)

**Comment by jonparsonsjncc** [comment]
Hi Andy

Could you let us know when the code is updated and pushed into the repo - as we have to then update the docker container

Thanks

Jon

Jon Parsons
Technical Delivery Manager
(DIP DigM)
JNCC, Monkstone House, City Road, Peterborough, PE1 1JY
Tel: 01733 562626, Direct Dial: 01733 866827, Mobile: 07526 497077
Email: ***@***.******@***.***>
My working times are: Monday to Friday 8.45am - 5pm (I may send emails outside of these hours but I do not expect a response outside your own standard working hours).
My pronouns are he/him

**Comment by jonparsonsjncc** [comment]
Hi @andyb-esdm @SimonAnnetts 
Just a quick check to see where we are with the upgrade
I notice in the thread you mention a staging server - we are presuming this is your staging server as we have totally redeveloped our staging environments
I also notice you mentioned that the code was updated - could you push this in to the master branch without needing to upgrading the server - if you could that would be great as we have everything here updated and ready for the new code and we are keen to deploy and test
Could you confirm
thanks
jon

**Comment by andyb-esdm** [comment]
@jonparsonsjncc Very sorry for the delay in my reply.
You're right, we are referring to the staging server. It doesn't affect your environment but we need to do that to test against the new API.
The code has been updated to .net core 3.1 and angular 11. I'm still to update the autocomplete and slider components. But that's a minor issue and isn't strictly needed (they worked okay when I tested locally).
Unfortunately, Simon is on leave this week and his changes are on a feature branch. They aren't big changes but we haven't done an integrated test against this API and the mapper. And I haven't confirmed with him they are complete (I _think_ they are).

I can merge but we should maybe discuss then you take it?

**Comment by HelenwoodsJNCC** [comment]
@SimonAnnetts can you give an update on the above?

**Comment by SimonAnnetts** [comment]
The changes I made are complete subject to testing against the mapper. I have left the changes in a feature branch while we update our staging server to dotnet core 3.1.
I've been away on leave for two weeks which is why I've not made any progress on this.
Once our server is updated then Andy can properly test his mapping code against the new dotnet core 3.1 API (I don't expect there to be any issues) and we can then merge the feature branch into master. I have scheduled this work on the server update for Thursday this week (July 8th).

thanks

**Comment by HelenwoodsJNCC** [comment]
@SimonAnnetts ok great thanks. If you can let myself and @jonparsonsjncc know how it goes tomorrow, that would be great.

**Comment by HelenwoodsJNCC** [comment]
@andyb-esdm will it be a simple enough job to test your mapping code against the new dotnet core? We are looking to do a deployment asap.

**Comment by andyb-esdm** [comment]
@HelenwoodsJNCC We're doing it today. Sorry for the delay but Simon is a lot more familiar with the infrastructure.
Bear in mind there are two branches:
master - the 'vanilla' mapper that you use for EMODnet
analytics - the mapper with google scripts added for MPA
I take it you will use our vscode task scripts to know how to compile the angular app. Let me know if you need any further details.
We'll let you know as soon as the repository is updated (and we have tested) today.

**Comment by HelenwoodsJNCC** [comment]
@andyb-esdm thanks!

**Comment by SimonAnnetts** [comment]
So the test server we have is now running dotnet core 3.1 and the updated web-api is running well on there now.

http://jnccdev-api.esdm.co.uk/api/MapInstance/emodnet

I'll now merge my feature branch into devel so that Andy can test against the mapper.

**Comment by SimonAnnetts** [comment]
sorry that should have read merge into master ....

**Comment by andyb-esdm** [comment]
@HelenwoodsJNCC both the master and analytics branches are now up-to-date and have been pushed to github. I did some quick testing around the main areas of functionality and all seems good.
Let us know how your deployment goes ...

**Comment by HelenwoodsJNCC** [comment]
Great, thanks @andyb-esdm @SimonAnnetts. Will let you know how things go at this end.

**Comment by HelenwoodsJNCC** [comment]
@andyb-esdm @SimonAnnetts - all seemed to go ok with the deployment here. No issues so far.

**Comment by andyb-esdm** [comment]
@HelenwoodsJNCC That's great. Thanks for letting us know.
